lib/lib-table-cluster/src/main/java/org/sagebionetworks/table/cluster/TableIndexDAOImpl.java

```diff
@@ -885,9 +886,19 @@ public ColumnAggregation mapRow(ResultSet rs, int rowNum)
 				ColumnModel model = new ColumnModel();
 				model.setName(aggregation.getColumnName());
 				ColumnType type = AnnotationType.valueOf(typeString).getColumnType();
+
+				//check if a LIST columnType needs to be used
+				if(aggregation.getListSize() > 1){
+					try {
+						type = ColumnTypeListMappings.forNonListType(type).getListType();

```

Comment:
optional might work better than illegal arg for getListType();

Reply:
Other parts of code is relying on it throwing an exception being thrown if unexpected usage of  conversion to list type occurrs